client/benchmark/src/main/java/org/elasticsearch/client/benchmark/AbstractBenchmark.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.client.benchmark;
+
+import org.elasticsearch.client.benchmark.ops.bulk.BulkBenchmarkTask;
+import org.elasticsearch.client.benchmark.ops.bulk.BulkRequestExecutor;
+import org.elasticsearch.client.benchmark.ops.search.SearchBenchmarkTask;
+import org.elasticsearch.client.benchmark.ops.search.SearchRequestExecutor;
+import org.elasticsearch.common.SuppressForbidden;
+
+import java.io.Closeable;
+
+public abstract class AbstractBenchmark<T extends Closeable> {
```

Comment:
wondering if the generics is really needed. The client is created by subclasses anyways, and the reason to create it here is so that we can close it? Maybe we could rather have an abstract close method that we call instead of needing the generic? Just thinking out loud, not a huge deal.